maestrowf/interfaces/script/localpooladapter.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import os
+
+from maestrowf.abstracts.interfaces import SchedulerScriptAdapter
+from maestrowf.abstracts.enums import JobStatusCode, State, SubmissionCode, \
+    CancelCode
+from maestrowf.interfaces.script import CancellationRecord, SubmissionRecord
+from pyaestro.utilities.executor import Executor, ExecTaskState, ExecCancel
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LocalPoolAdapter(SchedulerScriptAdapter):
+    """Interface class for the flux scheduler (on Spectrum MPI)."""
+
+    key = "local_pool"
+
+    def __init__(self, **kwargs):
+        """
+        Initialize an instance of the LocalPool Adapter.
+
+        The expected keyword arguments that are expected when the Flux adapter
+        is instantiated are as follows:
+        * num_processes: The number of local workers for processing.
+
+        :param **kwargs: A dictionary with default settings for the adapter.
+        """
+        super(LocalPoolAdapter, self).__init__(**kwargs)
+        num_workers = kwargs.get("num_workers", "1")
+        self.add_batch_parameter("max_workers", num_workers)
+        self._pool = Executor(num_workers)
+        self._extension = ".lp.sh"
+
+    @property
+    def extension(self):
+        return self._extension
+
+    def get_header(self, step):
+        """
+        Generate the header present at the top of Flux execution scripts.
+
+        :param step: A StudyStep instance.
+        :returns: A string of the header based on internal batch parameters and
+                  the parameter step.
+        """
+        return f"#!{self._exec}"
+
+    def get_parallelize_command(self, procs, nodes=None, **kwargs):
+        """
+        Generate an empty parallelize command for local execution.
+
+        :param procs: Number of processors to allocate to the parallel call.
+        :param nodes: Number of nodes to allocate to the parallel call
+                      (default = 1).
+        :returns: A string of the parallelize command configured using nodes
+                  and procs.
+        """
+        return ""
+
+    def submit(self, step, path, cwd, job_map=None, env=None):
+        """
+        Submit a script to the Flux scheduler.
+
+        :param step: The StudyStep instance this submission is based on.
+        :param path: Local path to the script to be executed.
+        :param cwd: Path to the current working directory.
+        :param job_map: A dictionary mapping step names to their job
+                        identifiers.
+        :param env: A dict containing a modified environment for execution.
+        :returns: The return status of the submission command and job
+                  identifier.
+        """
+        try:
+            job_id = self._pool.submit(path, cwd)
```

Comment:
there a future being tracked in here somewhere?

Reply:
This pool is entirely backed by `pyaestro` -- which does implement this with a future.